packages/video_player/video_player/lib/video_player.dart

```diff
@@ -763,6 +792,14 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     Future<ClosedCaptionFile>? closedCaptionFile,
   ) async {
     _closedCaptionFile = await closedCaptionFile;
+
+    _sortedCaptions = _closedCaptionFile?.captions;
+
+    /// Sort the captions by start time so that we can do a binary search.
+    _sortedCaptions?.sort((Caption a, Caption b) {
```

Comment:
`_updateClosedCaptionWithFuture` is called frequently. Re-sorting every caption in the entire file every time this method is called is likely going to hurt overall performance relative to the state before this PR.

The sorted captions should be cached, and only re-generated when the file is changed.

Reply:
I assumed _updateClosedCaptionWithFuture gets called only when file changes, and i cant find any other place it was called, but I made a method that that sorts the captions, and made calls to it when the file changes after `_updateClosedCaptionWithFuture`
I hope thats more efficient now.

Reply:
> I assumed _updateClosedCaptionWithFuture gets called only when file changes, and i cant find any other place it was called

https://github.com/flutter/packages/blob/a7c1159ac3814bdd4669156bed3f7d5692cece4a/packages/video_player/video_player/lib/video_player.dart#L511C44-L511C68

Reply:
noted, so my fix is not accurate, will think of other way and comment in this to let you know

Reply:
I hope the added check is good enough, I couldnt think of a better way, since we need to make sure it runs on every new file, do you think I should move back the sorting with the new check in _updateClosedCaptionWithFuture or leaving them seperated is cleaner.